### PR TITLE
Add Documentation to diagnostics.py File

### PR DESCRIPTION
**Pull request by GarethTMorgan**
# Pull Request
### Description
Adds class and method docstrings to ClockDiagnostic and GridDiagnostic classes. Also got rid of some whitespaces.

This pull request addresses #62

### CODE REVIEW

File: turbopy/diagnostics.py [hunk 1]
```diff
@@ -149,33 +150,53 @@ def initialize(self):
...
+        The 'Simulation' object that contains this object
+    input_data : dict
+        Dictionary containing information abut the class such as its name
+
+    Attributes
+    ----------
+    owner : Simulation
+        The 'Simulation' object that contains this object
+    input_data : dict
+        Dictionary containing information abut the class such as its name
```
Comment (arichar6) [inline turbopy/diagnostics.py:186]:
```suggestion
        Dictionary containing information about this diagnostic such as its name
```

File: turbopy/diagnostics.py [hunk 2]
```diff
@@ -184,19 +205,44 @@ def finalize(self):
...
 class ClockDiagnostic(Diagnostic):
+    """Diagnostic subclass used to store and save time data into a CSV file
+    using the CSVOutputUtility class.
+
+    Parameters
+    ----------
+    owner : Simulation
+        The 'Simulation' object that contains this object
+    input_data : dict
+        Dictionary containing information abut the class such as its name
```
Comment (arichar6) [inline turbopy/diagnostics.py:216]:
```suggestion
        Dictionary containing information about this diagnostic such as its name
```

File: turbopy/diagnostics.py [hunk 3]
```diff
@@ -184,19 +205,44 @@ def finalize(self):
...
+
+    Attributes
+    ----------
+    owner : Simulation
+        The 'Simulation' object that contains this object
+    input_data : dict
+        Dictionary containing information abut the class such as its name
```
Comment (arichar6) [inline turbopy/diagnostics.py:223]:
```suggestion
        Dictionary containing information about this diagnostic such as its name
```

File: turbopy/diagnostics.py [hunk 4]
```diff
@@ -184,19 +205,44 @@ def finalize(self):
...
+        Array to store values to be written into a CSV file
+    """
+
     def __init__(self, owner: Simulation, input_data: dict):
         super().__init__(owner, input_data)
         self.filename = input_data["filename"]
         self.csv = None
 
     def diagnose(self):
+        """Append time into the self.csv ndarray."""
```
Comment (arichar6) [inline turbopy/diagnostics.py:236]:
```suggestion
        """Append time into the csv buffer."""
```

File: turbopy/diagnostics.py [hunk 5]
```diff
@@ -184,19 +205,44 @@ def finalize(self):
...
         self.csv.append(self.owner.clock.time)
 
     def initialize(self):
+        """Initialize self.csv as an instance of the CSVOuputUtility class."""
```
Comment (arichar6) [inline turbopy/diagnostics.py:240]: I'm not sure if this is the correct notation, but can you add the code that that `CSVOutputUtility` automatically gets linked?

```suggestion
        """Initialize `self.csv` as an instance of the :class:`CSVOuputUtility` class."""
```

**Comment by GarethTMorgan** [comment]
Addressed and committed the changes you suggested @arichar6 . Thanks for the review 👍

**Pull request merged**